### ISSUE TITLE
The detailed mode for Phalcon/Debug is not work

### ISSUE THREAD

**Issue by limingxinleo** [phalcon/cphalcon#13315]
> **When I use Phalcon/Debug detailed mode,The result is not expected.**

```php
<?php
        $obj = new \stdClass();
        $obj->aaa = 111;

        array_map(function ($x) {
            $string = (new \Phalcon\Debug\Dump(null, true))->variable($x);

            echo(PHP_SAPI == 'cli' ? strip_tags($string) . PHP_EOL : $string);

        }, [$obj]);
```
> The Result

```
The argument is not initialized or iterable() code:0 in phalcon/debug/dump.zep line 215
#0 [internal function]: Phalcon\Debug\Dump->output(Object(stdClass), '')
#1 /Users/limx/Applications/limingxinleo/basic-phalcon/app/Tasks/TestTask.php(16): Phalcon\Debug\Dump->variable(Object(stdClass))
#2 [internal function]: App\Tasks\TestTask->App\Tasks\{closure}(Object(stdClass))
#3 /Users/limx/Applications/limingxinleo/basic-phalcon/app/Tasks/TestTask.php(20): array_map(Object(Closure), Array)
#4 [internal function]: App\Tasks\TestTask->mainAction(Array, Array)
#5 [internal function]: Phalcon\Cli\Dispatcher->callActionMethod(Object(App\Tasks\TestTask), 'mainAction', Array)
#6 [internal function]: Phalcon\Dispatcher->dispatch()
#7 /Users/limx/Applications/limingxinleo/basic-phalcon/vendor/limingxinleo/x-phalcon-console/src/Cli/XConsole.php(36): Phalcon\Cli\Console->handle(Array)
#8 /Users/limx/Applications/limingxinleo/basic-phalcon/run(47): Xin\Phalcon\Cli\XConsole->handle(Array)
#9 {main}%
```

I find the source code for Phalcon/Debug/Dump, using `for` instead of `each`, `for` is not support object, so it throw a exception, but `each` is deprecated for php7.2.

### Details

* Phalcon version: (`3.3.2`)
* PHP Version: (`7.1.14`)
* Operating System: MAC
* Installation type: Compiling from source || installing via package manager
* Zephir version (if any):
* Server: Nginx | Apache | Other
* Other related info (Database, table schema):

**Comment by sergeyklay** [comment]
cc: @jmsfwk

**Comment by sergeyklay** [comment]
Fixed in the `3.4.x` branch. Feel free to open a new issue if the problem appears again. Thank you for contributing.
